gdnative-derive/src/derive_macro.rs

```diff
@@ -48,13 +60,30 @@ pub(crate) fn parse_derive_input(input: TokenStream) -> DeriveData {
     };
 
     // read exported properties
-    if let Fields::Named(_names) = struct_data.fields {
-        // TODO
-    }
+    let props = if let Fields::Named(names) = &struct_data.fields {
+        names
+            .named
+            .iter()
+            .filter(|&field| {
+                field
+                    .attrs
+                    .iter()
+                    .filter(|attribute| attribute.path.segments[0].ident == "property")
```

Comment:
Please check the last segment's identifier, not the first: you want to match `gdnative::property`, not `property::foo::bar`.

Reply:
Sorry I was orientating myself on already existing code which also seemed to make use of the first segment. Maybe those lines should then be changed as well.

```
let user_data = input
        .attrs
        .iter()
        .find(|a| a.path.segments[0].ident == "user_data")
```

Reply:
They should, then. It would be great if you could cover those as well!

Reply:
I think we got a little bit confused here... because attribute.path.segments does match the identifier of the attribute macro and not the path of a passed class or type.

To avoid this confusion I suggest replacing `attribute.path.segements[0].ident == "name_of_attribute"` with the shorter check `attribute.path.is_ident("name_of_attribute")`

Reply:
Nobody got confused here. An attribute is named by a path, not just an identifier. An example of such a "scoped" attribute is `#[rustfmt::skip::macros(vec)]`.

However, this comment does remind me that simply checking the last segment is also incorrect. When the path isn't just a single identifier, we would also want to check if the preceding path is `gdnative::`.

I think we can proceed with checking the first segment for now, and fix all the checks at a later point in time.